architecture/2024-11-ucp-component-threat-model.md

```diff
@@ -0,0 +1,276 @@
+3. **Deployment Engine sends the list of resources back to the UCP as requests**
+4. **UCP sends each request to the corresponding Resource Provider**
+5. **UCP updates its data store with the information of the resources**
+
+### Threats
+
+#### Threat: Spoofing Deployment Engine Could Cause Information Disclosure and Denial of Service
+
+**Description:** If a malicious actor can spoof the Deployment Engine by tampering with the configuration in the UCP, the UCP will start sending requests to the malicious server. The malicious server can capture the traffic, leading to information disclosure. This would effectively disable the UCP, causing a Denial of Service.
+
+**Impact:** All data sent to the Deployment Engine by the UCP will be available to the malicious actor, including payloads of resources in the applications. The functionality of the UCP for managing resources will be disabled. Users will not be able to deploy updates to their applications.
+
+**Mitigations:**
+
+1. Tampering with the UCP code, configuration, or certificates would require access to modify the `radius-system` namespace. Our threat model assumes that the operator has limited access to the `radius-system` namespace using Kubernetes' existing RBAC mechanism.
+2. The resource payloads sent to the Deployment Engine by the UCP **MAY** contain sensitive operational information (e.g., passwords).
+
+**Status:** All mitigations listed are currently active. Operators are expected to secure their cluster and limit access to the `radius-system` namespace.
+
+#### Threat: Information Disclosure by Unauthorized Access to Secrets
+
+**Description:** A malicious actor could circumvent Kubernetes RBAC controls and gain unauthorized access to Kubernetes secrets managed by Radius. These secrets may contain sensitive information, such as credentials intended for use by applications.
+
+**Impact:** A malicious actor could gain access to sensitive information.
+
+**Mitigations:**
+
+1. Secret data managed by the UCP is stored at rest in Kubernetes secrets or etcd. Our threat model assumes that the API server and mechanisms like Kubernetes authentication/RBAC are not compromised.
+2. Secrets managed by Radius are always placed in the same namespace as the object that "owns" them. This is a requirement of the Kubernetes RBAC model.
+3. Secrets managed by Radius are subject to the Kubernetes RBAC model for controlling access. Operators are expected to limit access for users using existing tools.
+
+**Status:** All mitigations listed are currently active. Operators are expected to secure their cluster and limit access for users.
+
+#### Threat: Escalation of Privilege by Using Radius to Circumvent Kubernetes RBAC Controls
+
+**Description:** A malicious actor could circumvent Kubernetes RBAC controls and create arbitrary resources in Kubernetes by using Universal Control Plane. UCP operates with a wide scope of permissions in Kubernetes and the cloud.
```

Comment:
> UCP operates with a wide scope of permissions in Kubernetes and the cloud.

This is not actually true for UCP. Can you please review the permissions that UCP has in Kubernetes and update this?